keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt

```diff
@@ -0,0 +1,36 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.igor.BuildService
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactMetadata
+import com.netflix.spinnaker.keel.api.artifacts.BuildMetadata
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import kotlinx.coroutines.runBlocking
+import org.springframework.stereotype.Component
+
+/**
+ * Provides functionality to convert build metadata, which is coming from internal service, to artfiact metadata (via igor).
+ */
+@Component
+class ArtifactMetadataService(
+  private val buildService: BuildService
+) {
+
+  fun getArtifactMetadata(
+    commitId: String?,
+    buildNumber: String?
+  ): ArtifactMetadata? {
```

Comment:
Generally the pattern I see in other services is to have non-nullable parameters to these calls, and have the caller handle nullability, as it doesn't really make sense to make the call in the first place if you don't have those parameters available.

Reply:
Do you think it'll be better to check the nullability prior to making this call? I can move it to the actual suppliers.

Reply:
Yes, that's what I was trying to say.

Reply:
done!